orchestrator.py

```diff
@@ -0,0 +1,131 @@
+import json
+import io
+import random
+import re
+import struct
+import time
+import traceback
+from datetime import datetime
+from queue import Queue
+from queue import Empty
+
+from services.mock_ai_service import MockAIService
+
+from scenes.translator_scene import TranslatorScene
+
+from threading import Thread
+
+class Orchestrator():
+    def __init__(self, image_setter, microphone, ai_tts_service, ai_image_gen_service, ai_llm_service, story_id, language):
+        self.image_setter = image_setter
+        self.microphone = microphone
+
+        self.ai_tts_service = ai_tts_service
+        self.ai_image_gen_service = ai_image_gen_service
+        self.ai_llm_service = ai_llm_service
+
+        
+        self.language = language
+
+        self.tts_getter = None
+        self.image_getter = None
+        self.llm_response_thread = None
+
+        self.scene_queue = Queue()
+        self.stop_threads = False
+
+    def handle_user_speech(self, message):
+        # TODO Need to support overlapping speech here!
+        print(f"👅 Handling user speech: {message}")
+        Thread(target=self.request_llm_response, args=(message,)).start()
+
+    def request_llm_response(self, message):
+        try:
+            msgs = [{"role": "system", "content": f"You will be provided with a sentence in English, and your task is to translate it into {self.language.capitalize()}."}, {"role": "user", "content": message['text']}]
+            message['response'] = self.ai_llm_service.run_llm(msgs)
+            self.handle_translation(message)
+        except Exception as e:
+            print(f"Exception in request_llm_response: {e}")
+
+    def handle_translation(self, message):
+        # Do this all as one piece, at least for now
+        llm_response = message['response']
+        out = ''
+        for chunk in llm_response:
+            if len(chunk["choices"]) == 0:
```

Comment:
You're accessing `chunk["choices"]` a lot here - suggest setting this to a variable once up above and reusing. Then also using that variable as a base to set `chunk["choices"][0]["delta"]["content"]`, as you are also retrieving that twice below.

Aside: is it guaranteed that the response will contain all of these elements? If not, should prepare for potential exception.